inc/Action/Search.php

```diff
@@ -39,8 +39,8 @@ public function preProcess()
         if ($ID !== $conf['start'] && !$INPUT->has('q')) {
             parse_str($INPUT->server->str('QUERY_STRING'), $urlParts);
             $urlParts['q'] = $urlParts['id'];
-            $urlParts['id'] = $conf['start'];
-            $url = DOKU_URL . DOKU_SCRIPT . '?' . http_build_query($urlParts, null, '&');
+            unset($urlParts['id']);
+            $url = wl() . '?' . http_build_query($urlParts, null, '&');
```

Comment:
```suggestion
            $url = wl($ID, $urlParts, true);
```

Reply:
also, shouldn't the [`$conf['canonical']`](https://www.dokuwiki.org/config:canonical) setting be obeyed ?
```suggestion
            $url = wl('', $urlParts, $conf['canonical'] );
```

Reply:
This is a one-time redirect, so it doesn't matter whether the path is relative or absolute. And I prefer the absolute version in redirect.

Actually this is very inconsistent inside DokuWiki.

Reply:
$ID should be preserved, try this:

1. https://www.dokuwiki.org/non_exists
2. Search on the top of this page
3. Look at the URL: https://www.dokuwiki.org/start?do=search&id=non_exists&q=test
4. Submit in the inline search form: https://www.dokuwiki.org/non_exists?do=search&id=non_exists&sf=1&q=test

Reply:
Redirects should always be absolute. Most browsers understand relative redirects, but it's not RFC conform. `wl()` already honors the `$conf['canonical']` setting unless an absolute url is requested via the third parameter. Since this is not output to HTML, the fourth parameter should be set to `&`;

```suggestion
            $url = wl($ID, $urlParts, true, '&');
```